lang/cs/Org.Apache.REEF.Network/Elastic/Topology/Logical/Impl/DataNode.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Network.Elastic.Topology.Logical.Enum;
+using Org.Apache.REEF.Utilities.Attributes;
+using System.Collections.Generic;
+
+namespace Org.Apache.REEF.Network.Elastic.Topology.Logical.Impl
+{
+    /// <summary>
+    /// Represents a node in the operator topology graph.
+    /// Nodes are logical representations in the Driver for tasks.
+    /// </summary>
+    [Unstable("0.16", "API may change")]
+    internal sealed class DataNode
+    {
+        private readonly int _taskId;
+        private readonly bool _isRoot;
+        private readonly List<DataNode> _children;
+
+        private DataNode _parent;
+        private DataNodeState _state;
+
+        /// <summary>
+        /// Construct a node using a given task id.
+        /// </summary>
+        /// <param name="taskId">The id for the node</param>
+        /// <param name="isRoot">Whether the node is the root/master of the topology or not</param>
+        public DataNode(
+            int taskId,
+            bool isRoot)
+        {
+            _taskId = taskId;
+            _isRoot = isRoot;
+            _state = DataNodeState.Reachable;
+
+            _children = new List<DataNode>();
+        }
+
+        /// <summary>
+        /// The current state for the node.
+        /// </summary>
+        public DataNodeState FailState
+        {
+            get { return _state; }
+            set { _state = value; }
+        }
+
+        /// <summary>
+        /// The parent of the target node.
+        /// </summary>
+        public DataNode Parent
+        {
+            get { return _parent; }
+            set { _parent = value; }
+        }
+
+        /// <summary>
+        /// Add a node to the list of children nodes of the current one.
+        /// </summary>
+        public void AddChild(DataNode child)
+        {
+            _children.Add(child);
+        }
```

Comment:
if we implement
```C#
public void AddChildren(IEnumerable<DataNode> children)
{
    _children.AddRange(children);
}
```
things will get a bit more elegant on teh client side

Reply:
Fixed.